### PR TITLE
Hurumap data plugin page

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Desktop Screenshots

localhost_8080_wp-admin_admin php_page=hurumap-data (4)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: wp-content/plugins/hurumap-data/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["@babel/react", ["@babel/env", { "modules": false }]]
+}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/.babelrc]: `/n`

File: wp-content/plugins/hurumap-data/api.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        array(
+            'methods'               => 'POST',
+            'callback'              => 'update_or_create_chart'
+        ),
+    ));
+    $endpoint_sections = '/sections';
+    register_rest_route($namespace, $endpoint_sections, array(
+        array(
+            'methods'               => 'POST',
+            'callback'              => 'update_or_create_sections'
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/api.php:29]: `..._create_chart` && `...__create_section` **OR** `..._create_charts` && `..._create_sections`

File: wp-content/plugins/hurumap-data/api.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    if ($published === null) {
+        $hurumap = $wpdb->get_results("SELECT * FROM {$wpdb->base_prefix}hurumap_charts order by created_at desc");
+        $flourish = $wpdb->get_results("SELECT * FROM {$wpdb->base_prefix}flourish_charts order by created_at desc");
+        $sections = $wpdb->get_results("SELECT * FROM {$wpdb->base_prefix}chart_sections order by created_at desc");
+    } else {
+        $published = $published  === '1';
+
+        $hurumap = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}hurumap_charts where published = %d order by created_at desc", $published));
+        $flourish = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}flourish_charts where published = %d order by created_at desc", $published));
+        $sections = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}chart_sections where published = %d order by created_at desc", $published));
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/api.php]: QQ: Does MySQL support `IN` queries? If it does, we don't need `if/else`, we can just have `IN [1]` for published and `IN [0,1]` for both draft & published, etc.

File: wp-content/plugins/hurumap-data/db.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                `title`      varchar(255) NOT NULL ,
+                `subtitle`   varchar(255) NOT NULL ,
+                `file`       json NOT NULL ,
+                `published` tinyint NOT NULL DEFAULT 0,
+                `created_at` datetime NOT NULL DEFAULT NOW() ON UPDATE NOW(),
+                `updated_at` datetime NOT NULL DEFAULT NOW(),
+
+                PRIMARY KEY (`id`),
+                KEY `fkIdx_22` (`section`),
+                CONSTRAINT `FK_22` FOREIGN KEY `fkIdx_22` (`section`) REFERENCES `{$wpdb->base_prefix}chart_sections` (`id`)
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/db.php:33]: Dumb question, what's up with the numbers 22, 19, etc.?

File: wp-content/plugins/hurumap-data/index.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+function load_admin_scripts()
+{
+    $screen = get_current_screen();
+    if ($screen->id == 'toplevel_page_hurumap-data') {
+        wp_enqueue_script( 'hurumap-data-admin-script' );
+    }
+}
+
+function setup_admin_menu() {
+  add_menu_page('Hurumap Data', 'Hurumap Data', 'manage_options', 'hurumap-data', 'hurumap_data_root');
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/index.php]: ~Hurumap~ HURUmap

File: wp-content/plugins/hurumap-data/src/App.js [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+import gql from 'graphql-tag';
+
+import { Formik, FieldArray } from 'formik';
+import { Grid, Button } from '@material-ui/core';
+
+// import AceEditor from "react-ace";
+
+// import 'brace';
+// import 'brace/mode/json';
+// import 'brace/theme/github';
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/src/App.js]: ?

File: wp-content/plugins/hurumap-data/src/App.js [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+    'aria-controls': `simple-tabpanel-${index}`
+  };
+}
+
+function TabPanel(props) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      component="div"
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/src/App.js]: QQ: Why do we need `component="div"` in a `div`?

File: wp-content/plugins/hurumap-data/src/ChartFactory.js [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+                width: '400px',
+                height: '300px'
+              }}
+            >
+              <BarChart
+                key={key}
+                responsive
+                offset={45}
+                barWidth={barWidth || 40}
+                width={400 || primaryData.length * 2 * ((barWidth || 40) + 5)}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/src/ChartFactory.js]: As I mentioned, Dominion.AFRICA has a better `ChartFactory` than the Takwimu one e.g. it allows chart customization using `props` as well as calculate and sets height/width taking `padding`, `offset`, etc. into consideration: https://github.com/CodeForAfrica/Dominion.AFRICA/blob/5d9de33cc42b5741afd63f97d0e9d831f2579a63/src/components/ChartFactory.js#L244-L265

File: wp-content/plugins/hurumap-data/index.php [hunk 9]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * Plugin Name: HURUmap Data
+ * Plugin URI: https://github.com/CodeForAfrica/HURUmap-Data-WP
+ * Description: Dummy block for HURUmap UI WordPress plugin.
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/index.php]: Still _dummy_?

**Review by kilemensi — COMMENTED** [review]
:rocket: @karimkawambwa :

1. Can we have configs (graphql URL, api URLs, etc.) in the `config.js`,
2. If we're doing a full preview (numbervisuals, insight, etc.), it _might_ be better to have the preview below the knobs rather than on the side... especially on `md` screen,
3. Considerations:
  i. Key chart props like `horizontal` are not configurable,
  ii. Organize `level1`: They don't appear to be organized on my end. Either lets have level1 geos below their respective countries i.e. Tanzania, followed by level1 regions in Tanzania then Senegal, followed by level1 geos in Senegal, etc., or prepend/append country name to level1 e.g. `Dakar | Senegal`, etc.,

So yeah, a bit of clean up & I think this is good to go...

File: wp-content/plugins/hurumap-data/db.php (hunk 4 above)
Comment (karimkawambwa) [reply wp-content/plugins/hurumap-data/db.php:33]: Random generator

**Comment by karimkawambwa** [comment]
@kilemensi I suggest this for a separate pr since it could be something set to be configurable in wordpress settings.
> Can we have configs (graphql URL, api URLs, etc.) in the config.js,

**Comment by karimkawambwa** [comment]
@kilemensi @KhadijaMahanga @nerdlet 👀

File: wp-content/plugins/hurumap-data/api.php [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+    global $wpdb;
+    $json = $request->get_json_params();
+    foreach ($json as $section) {
+        if (!$wpdb->update(
+            "{$wpdb->base_prefix}chart_sections",
+            array('id' => $section['id'], 'name' => $section['name'], 'description' => $section['description']),
+            array('id' => $section['id'])
+        )) {
+            $wpdb->insert("{$wpdb->base_prefix}chart_sections", array('id' => $section['id'], 'name' => $section['name'], 'description' => $section['description']));
+        }
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/api.php:58]: Still no `UPSERT`? https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html

File: wp-content/plugins/hurumap-data/src/App.js [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+                      actions={[
+                        {
+                          label: 'Publish Selected',
+                          value: 'publish:selected'
+                        },
+                        { label: 'Publish All', value: 'publish:all' },
+                        { label: 'Delete Selected', value: 'delete:selected' },
+                        { label: 'Delete All', value: 'delete:all' }
+                      ]}
+                    /> */}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/src/App.js:255]: ?

File: wp-content/plugins/hurumap-data/.eslintrc.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+    rules: {
+      'no-unused-vars': 'error',
+      'react-hooks/rules-of-hooks': 'error',
+      'react-hooks/exhaustive-deps': 'warn',
+      'react/jsx-filename-extension': [1, { extensions: ['.js'] }],
+      'react/prop-types': [2, { ignore: ['classes'] }],
+      'react/jsx-props-no-spreading': 0
+    }
+  };
+
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/.eslintrc.js:18]: `\n`

File: wp-content/plugins/hurumap-data/package.json [hunk 13]
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "blocky",
```
Comment (kilemensi) [inline wp-content/plugins/hurumap-data/package.json]: We `hurumap-data` now

**Review by kilemensi — COMMENTED** [review]
Looks good @karimkawambwa :+1:  (ignoring the size _responsivity_ issues) but:

1.The aggregate stuff either doesn't work or _I don't get them_;
Screenshot from 2019-10-17 16-18-49
I've changed that to `%`, I expect the chart to show me **%** as well but it doesn't? Is it a label/tooltip issue or?
2. On the first load, the chart defaults to horizontal even though the toggle is off
Screenshot from 2019-10-17 16-34-29

**Comment by karimkawambwa** [comment]
@kilemensi 
> 2. On the first load, the chart defaults to horizontal even though the toggle is off

This might have to be due to the wrong default https://github.com/CodeForAfrica/HURUmap-UI/blob/master/src/factory/ChartFactory.js#L23

Fixing it there should fix it all.

File: wp-content/plugins/hurumap-data/src/App.js (hunk 11 above)
Comment (karimkawambwa) [reply wp-content/plugins/hurumap-data/src/App.js:255]: TODO @kilemensi

**Review by KhadijaMahanga — APPROVED** [review]
@karimkawambwa  LGTM  :fire: :fire:

File: wp-content/plugins/hurumap-data/api.php [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+    return $response;
+}
+
+function delete_charts($request)
+{
+    global $wpdb;
+
+    $json = $request->get_json_params();
+    $placeholders = implode(', ', array_fill(0, count($json), '%s'));
+    $success = $wpdb->query($wpdb->prepare("DELETE FROM {$wpdb->base_prefix}chart_charts WHERE id IN ({$placeholders})", $json));
```
Comment (KhadijaMahanga) [inline wp-content/plugins/hurumap-data/api.php]: `{$wpdb->base_prefix}chart_charts` do we have a `chart_charts` table?

**Pull request merged**